geo/src/algorithm/haversine_closest_point.rs

```diff
@@ -0,0 +1,579 @@
+        let line = Line::new(p_2, p_1);
+
+        let p_from = Point::new(-84.75625, 31.81056);
+        if let Closest::SinglePoint(pt) = line.haversine_closest_point(&p_from) {
+            assert!(pt.approx_eq(&Point::new(-85.13337428852164, 32.45365659858937)));
+        } else {
+            panic!("Did not get Closest::SinglePoint!");
+        }
+
+        let p_from = Point::new(-85.67211, 32.39774);
+        if let Closest::SinglePoint(pt) = line.haversine_closest_point(&p_from) {
+            assert!(pt.approx_eq(&Point::new(-85.58999680564376, 32.26023534389268)));
+        } else {
+            panic!("Did not get Closest::SinglePoint!");
+        }
+    }
+
+    #[test]
+    fn point_to_line_intersection() {
+        let p_1 = Point::new(-84.74905, 32.61454);
+        let p_2 = Point::new(-85.93942, 32.11055);
+        let line = Line::new(p_2, p_1);
+
+        if let Closest::Intersection(pt) = line.haversine_closest_point(&p_1) {
+            assert!(pt == p_1);
+        } else {
+            panic!("Did not get Closest::Intersection!");
+        }
+    }
+
+    #[test]
+    fn point_to_line_intersection_2() {
+        let p_1 = Point::new(-84.74905, 32.61454);
+        let p_2 = Point::new(-85.93942, 32.11055);
+        let line = Line::new(p_2, p_1);
+
+        let p_from = Point::new(-85.13337428852164, 32.45365659858937);
+        if let Closest::Intersection(pt) = line.haversine_closest_point(&p_from) {
+            assert!(pt.approx_eq(&p_from));
+        } else {
+            panic!("Did not get Closest::Intersection!");
+        }
+    }
+
+    // Across the pole
+    #[test]
+    fn point_to_line_across_equator() {
+        let p_1 = Point::new(-38.424_794_871_794_916, 75.137_388_461_538_48);
+        let p_2 = Point::new(-28.608_712_820_512_863, -85.278_057_692_307_67);
+        let line = Line::new(p_2, p_1);
+        let p_from = Point::new(-25.86062, -87.32053);
+
+        if let Closest::SinglePoint(pt) = line.haversine_closest_point(&p_from) {
+            assert!(pt.approx_eq(&Point::new(-28.608_712_820_512_864, -85.278_057_692_307_67)));
+        } else {
+            panic!("Did not get Closest::SinglePoint!");
+        }
+    }
+
+    #[test]
+    fn point_to_line_across_close_to_north_pole() {
+        let p_1 = Point::new(-37.244_921_874_999_99, 79.508_612_500_000_03);
+        let p_2 = Point::new(50.596_875_000_000_01, 81.054_628_125_000_02);
+        let line = Line::new(p_2, p_1);
+        let p_from = Point::new(8.15172, 77.40041);
+
+        if let Closest::SinglePoint(pt) = line.haversine_closest_point(&p_from) {
+            assert!(pt.approx_eq(&Point::new(5.481_094_923_165_54, 82.998_280_987_615_33)));
+        } else {
+            panic!("Did not get Closest::SinglePoint!");
+        }
+    }
+
+    #[test]
+    fn point_to_linestring() {
+        let wkt = "LineString (3.86503906250000284 11.71231367187503736, 9.48691406250000568 17.3341886718750402, \
```

Comment:
This is not a big deal, so feel free to leave it as is, but since WKT disregards extra whitespace, you can omit the `\` and just have a multiline string.

This makes it slightly easier on yourself to write and for anyone wanting to copy/paste this WKT into another GIS engine

Reply:
Done.